integrated_channels/integrated_channel/management/commands/__init__.py

```diff
@@ -0,0 +1,96 @@
+class IntegratedChannelCommandMixin(object):
+    """
+    Contains common functionality for the IntegratedChannel management commands.
+    """
+    integrated_channel_choices = {
+        'sap': SAPSuccessFactorsEnterpriseCustomerConfiguration,
+    }
+
+    def add_arguments(self, parser):
+        """
+        Adds the optional arguments: ``--enterprise_customer``, ``--channel``
+        """
+        parser.add_argument(
+            '--enterprise_customer',
+            dest='enterprise_customer',
+            default=None,
+            metavar='ENTERPRISE_CUSTOMER_UUID',
+            help='Transmit data for only this EnterpriseCustomer. '
+            'Omit this option to transmit to all EnterpriseCustomers with active integrated channels.'
+        )
+        parser.add_argument(
+            '--channel',
+            dest='channel',
+            default=None,
+            metavar='INTEGRATED_CHANNEL',
+            help='Transmit data to this IntegrateChannel. '
+            'Omit this option to transmit to all configured, active integrated channels.',
+            choices=self.integrated_channel_choices.keys(),
+        )
+
+    def get_integrated_channels(self, options):
+        """
+        Generates a list of active integrated channels, filtered from the given options.
+
+        Raises errors when invalid options are encountered.
+
+        See ``add_arguments`` for the accepted options.
+        """
+        # If a valid enterprise customer uuid was provided, transmit for that customer.
+        # Otherwise, transmit for all customers.
+        enterprise_customer = None
+        enterprise_customer_uuid = options['enterprise_customer']
+        if enterprise_customer_uuid:
+            try:
+                enterprise_customer = EnterpriseCustomer.active_customers.get(uuid=enterprise_customer_uuid)
+            except EnterpriseCustomer.DoesNotExist:
+                raise CommandError('Enterprise customer {} not found, or not active'.format(enterprise_customer_uuid))
+
+        # Assemble a list of integrated channel classes to transmit to.
+        # If a valid channel type was provided, use it.
+        # Otherwise, use all the available channel types.
+        channel_code = options['channel']
+        if channel_code is not None:
+            if channel_code not in self.integrated_channel_choices:
+                raise CommandError('Invalid integrated channel: {}'.format(channel_code))
+            channel_classes = [self.integrated_channel_choices.get(channel_code)]
+        else:
+            channel_classes = self.integrated_channel_choices.values()
+
+        # Loop through each channel class (optionally for a specific enterprise customer)
+        for channel_class in channel_classes:
+            # Use Active channels only
+            integrated_channels = channel_class.objects.filter(active=True)
+            if enterprise_customer:
+                integrated_channels.filter(enterprise_customer=enterprise_customer)
```

Comment:
I think you need to assign this to a variable - if you just do `.filter()` on a queryset, it doesn't change the original item.